File: schemas/helpdesk-customer-support/drizzle/ticket_priorities.ts
// ticket_priorities: configurable priority levels with sort order and color.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";

export const ticketPriorities = pgTable(
  "ticket_priorities",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    slug: text("slug").notNull().unique(),
    sortOrder: integer("sort_order").notNull().default(0),
    color: text("color"),
    isDefault: boolean("is_default").notNull().default(false),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_ticket_priorities_sort_order").on(table.sortOrder),
  ]
);

File: schemas/helpdesk-customer-support/spacetimedb/ticket_priorities.rs
// ticket_priorities: Urgency levels for tickets.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = ticket_priorities, public)]
pub struct TicketPriority {
    #[primary_key]
    pub id: String, // UUID
    pub name: String,
    #[unique]
    pub slug: String,
    #[index(btree)]
    pub sort_order: i32,
    pub color: Option<String>,
    pub is_default: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
